vehicle/mkv/software/dashboard/dashboard.c

```diff
@@ -1,521 +1,159 @@
-			gFlag &= ~_BV(PRECHARGE);
-			gFlag &= ~_BV(STATUS_START);
-			gCAN_MSG[1] = 0x00;
-		}
-
-		if(can_recv_msg[4] == 0xFF) {
-			PORT_HV_LED |= _BV(HV_LED); //if the HV Check is High then turn on HV LED
-		}
-		else {
-			PORT_HV_LED &= ~_BV(HV_LED); //if the HV Check is 0 then turn the HV LED off
-		}
-
-		// If IMD shutdown is true, make IMD_PIN high (if IMD goes low within 2 seconds of car on)
-		// make sure these latch (don't turn off until board is turned off)
-
-		//Setup to Receive Again
-		CANSTMOB = 0x00;
-		CAN_wait_on_receive(AIR_CONTROL_CRITICAL_MBOX, CAN_ID_AIR_CONTROL_CRITICAL, CAN_LEN_AIR_CONTROL_CRITICAL, CAN_MSK_SINGLE);
-	}
-
-
-/*----- Throttle Mailbox -----*/
-	CANPAGE = (THROTTLE_MBOX << MOBNB0);
-	if(bit_is_set(CANSTMOB, RXOK)) {
-		can_recv_msg[0] = CANMSG;   // PANIC PANIC
-		can_recv_msg[1] = CANMSG;   // Throttle Position
-		can_recv_msg[2] = CANMSG;   // BOTs sense
-		can_recv_msg[3] = CANMSG;   // inertia Switch Sense
-		can_recv_msg[4] = CANMSG;   // Driver E-Stop Sense
-
-		OCR1A = can_recv_msg[1]; //Throttle for PWM
-
-
-		//Setup to Receive Again
-		CANSTMOB = 0x00;
-		CAN_wait_on_receive(THROTTLE_MBOX, CAN_ID_THROTTLE, CAN_LEN_THROTTLE, CAN_MSK_SINGLE);
-	}
-
-
-}
-
-ISR(PCINT1_vect) { //pin inturput, me telling myself something, stepping on a lego and my inclination to say owwww
-
-	if(bit_is_clear(PINC,START_PIN)) { //Pull up resistor. Therefore this should be bit is clear
-		gFlag |= _BV(STATUS_START);
-		PORT_DEBUG_LED2 |= _BV(DEBUG_LED2);
-	} else {
-		gFlag &= ~_BV(STATUS_START);
-		PORT_DEBUG_LED2 &= ~_BV(DEBUG_LED2);
-	}
-}
-
-/*ISR(PCINT2_vect){
-	if(bit_is_set(PIND,PD2)) {
-		PORTC |= _BV(PC0);
-	}else{
-		PORTC &= ~_BV(PC0);
-	}
-} */
-
-ISR(TIMER0_COMPA_vect) {
-	// Only send CAN msgs every 20 cycles
-	// Timer Counter0 compare match A
- gTimerFlag |= _BV(UPDATE_STATUS);
- }
-
-	if(gClock_prescale > 10) {
-		gFlag |= _BV(UPDATE_STATUS);
-		gClock_prescale = 0;
-	}
-	gClock_prescale++;
+    Header:
+    Contains code for the dashboard board
+    Function: Responsible for AMS and IMD LED Indicators - DONE
+                -listening for CAN messages and controlling LED appropriately
+            : Start button + LED - DONE
+                - based on input off of the physical start button before entering RTD
+            : Interface with LED Bars Boards - TODO
+                - listening for CAN messages and outputting over I2C
+            : Implement other LEDs (brake, LV, HV) - DONE
+                - CAN
+            : Implement power cycle logic - DONE
+            : Implement RTD Buzzer - DONE
+
+
+    Questions
+*/
+
+#include "dashboard.h"
+#include "libs/can/api.h"
+#include "libs/gpio/api.h"
+#include "libs/timer/api.h"
+#include "libs/adc/api.h"
+
+volatile bool START_BUTTON_STATE;
+volatile bool HV_STATE;
+volatile bool BRAKE_PRESSED;
+volatile bool READYTODRIVE;
+
+volatile bool send_can;
+
+volatile int buzzer_counter = 0;
+int steering_pos;
+
+
+// CAN Interrupts
+ISR(CAN_INT_vect){
```

Comment:
This interrupt can be replaced with `can_receive` functions (and we shouldn't check for CAN in interrupts--it's bad practice). But anyways we're gonna replace it with the new CAN API so it's not a big deal. Just wanted to make a note.